service/src/server/routes/subgraphs.rs

```diff
@@ -1,55 +1,74 @@
-
 use axum::{
     body::Bytes,
+    extract::Extension,
     http::{self, HeaderName, Request, StatusCode},
     response::IntoResponse,
     Json,
 };
-use tracing::info;
+use std::sync::Arc;
+use tracing::trace;
 
 use crate::{
     query_processor::{FreeQuery, SubgraphDeploymentID},
     server::ServerOptions,
 };
 
 pub async fn subgraph_queries(
-    req: Request<axum::body::Body>,
-    server: axum::extract::Extension<ServerOptions>,
+    Extension(server): Extension<ServerOptions>,
     id: axum::extract::Path<String>,
-    query: Bytes,
+    req: Request<axum::body::Body>,
 ) -> impl IntoResponse {
-    info!("got request");
-    let query_string = String::from_utf8_lossy(&query);
-    // Initialize id into a subgraph deployment ID
-    let subgraph_deployment_id = SubgraphDeploymentID::new(Arc::new(id).to_string());
-
     // Extract scalar receipt from header and free query auth token for paid or free query
-    let _receipt = if let Some(recipt) = req.headers().get("scalar-receipt") {
+    let receipt = if let Some(recipt) = req.headers().get("scalar-receipt") {
         match recipt.to_str() {
             Ok(r) => Some(r),
             Err(_) => {
                 let error_body = "Bad scalar receipt for subgraph query".to_string();
-                return (StatusCode::BAD_REQUEST, Json(error_body)).into_response();
+                return (
+                    StatusCode::BAD_REQUEST,
+                    axum::response::AppendHeaders([
+                        (axum::http::header::CONTENT_TYPE, "application/json"),
+                        (HeaderName::from_static("graph-attestable"), "false"),
+                    ]),
+                    Json(error_body),
+                );
             }
         }
     } else {
         None
     };
+    trace!(
+        "receipt attached by the query, can pass it to TAP: {:?}",
+        receipt
+    );
 
     // Extract free query auth token
     let auth_token = req
         .headers()
         .get(http::header::AUTHORIZATION)
         .and_then(|t| t.to_str().ok());
-    let mut free = false;
     // determine if the query is paid or authenticated to be free
-    if auth_token.is_some()
+    let free = auth_token.is_some()
         && server.free_query_auth_token.is_some()
-        && auth_token.unwrap() == server.free_query_auth_token.as_deref().unwrap()
-    {
-        free = true;
-    }
+        && auth_token.unwrap() == server.free_query_auth_token.as_deref().unwrap();
+
+    let query: Bytes = hyper::body::to_bytes(req.into_body())
+        .await
+        .map_err(|e| {
+            (
+                StatusCode::BAD_REQUEST,
+                axum::response::AppendHeaders([
+                    (axum::http::header::CONTENT_TYPE, "application/json"),
+                    (HeaderName::from_static("graph-attestable"), "false"),
+                ]),
+                Json(e),
+            )
+        })
+        .unwrap();
+    let query_string = String::from_utf8_lossy(&query);
+
```

Comment:
`.unwrap()`s are still there, and will halt the program instead or returning an error HTTP response, or is there something I am misunderstanding?

Reply:
You are right that the program will not return an HTTP response, it will just return `curl: (52) Empty reply from server` if I do a simple curl with malformed query. 
I was trying to avoid the long match/combinator statements, but now I've refactored so that program will return an error HTTP response, think it is much better:)
like this
```
"Indexer error: IE075, explanation: Failed to parse response body to query string, cause: IndexerErrorCause(FromUtf8Error { bytes: [240, 159, 144], error: Utf8Error { valid_up_to: 0, error_len: None } })
```



Note: this now outdated iteration of resolver function will indeed lead to a panick thread, but the program will not halt. The server would log something like below and continue serving future requests
```
thread 'tokio-runtime-worker' panicked at 'called `Result::unwrap()` on an `Err` value: (400, Json(FromUtf8Error { bytes: [240, 159, 144], error: Utf8Error { valid_up_to: 0, error_len: None } }))', service/src/server/routes/subgraphs.rs:71:10
```